tests/integration-tests/conftest.py

```diff
@@ -866,6 +867,35 @@ def key_name(request):
     return request.config.getoption("key_name")
 
 
+@pytest.fixture()
+def amis_dict(request, architecture, region):
+    """
+    Get the amis dict in cloudforamtion template for given region and architecture.
+
+    In released test, ami dict contains released public amis with different OSes
+    In develop test, ami dict contains newly built private amis with different OSes
+    :return: amis dict
+    """
+    template_url = request.config.getoption("template_url")
+    if template_url:
```

Comment:
can we read the amis from the official template in case template_url is not set so that amis_dict is always returning the amis and we don't need to handle the extra boto3 call outside?

Reply:
done